### PR TITLE
Image Customizer: Check for file system errors.

### PR DESCRIPTION
**Pull request by cwize1**
After customization is complete, check for any file system errors and fail if any are found.

---

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

###### Does this affect the toolchain?  

NO

###### Test Methodology

- Manually ran image customizer tool.

### CODE REVIEW

File: toolkit/tools/pkg/imagecustomizerlib/checkfilesystem.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+		if diskPartition.Type != "part" {
+			continue
+		}
+
+		// Check the file system.
+		// Note: If file systems errors are found and corrected, `fsck -a` will return an exit code of 1, which is
+		// currently still treated as a failure.
+		err := shell.ExecuteLive(true /*squashErrors*/, "fsck", "-a", diskPartition.Path)
+		if err != nil {
+			return fmt.Errorf("failed to check (%s) with fsck:\n%w", diskPartition.Path, err)
```
Comment (romoh) [inline toolkit/tools/pkg/imagecustomizerlib/checkfilesystem.go:54]: Is there any info we would want to capture here when that happens? Should we run a couple of extra commands for debugging?

Comment (cwize1) [reply toolkit/tools/pkg/imagecustomizerlib/checkfilesystem.go:54]: I'm not sure what would help with debugging a corrupted disk.

Comment (romoh) [reply toolkit/tools/pkg/imagecustomizerlib/checkfilesystem.go]: I'm now thinking maybe we should let the fsck command run on all partitions and fail afterwards, that way we tell which ones failed the check (if not all)

**Review by romoh — APPROVED** [review]

File: toolkit/tools/pkg/imagecustomizerlib/checkfilesystem.go (hunk 1 above)
Comment (cwize1) [reply toolkit/tools/pkg/imagecustomizerlib/checkfilesystem.go]: Sure, fixed.

**Pull request merged**